Chap_Terms.tex

```diff
@@ -4,39 +4,30 @@
+\item \declareterm{slot}\declareterm{slots}\emph{slot} refers to an allocated set of resources assigned to a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or cooperatively at any given time during their execution.
+\item \declareterm{job}\emph{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a \emph{session}. For example, ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' is considered a single \ac{MPMD} job containing two applications.
 \item \declareterm{namespace}\emph{namespace} refers to a character string value assigned by the \ac{RM} to a \textit{job}. All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM}.
-\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or in parallel at any given time during their execution.
-\item \declareterm{rank}\emph{rank} refers to the numerical location (starting from zero) of a process within the defined scope. Thus, {global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
-\item \declareterm{workflow}\declareterm{workflows}\emph{workflow} refers to an orchestrated execution plan frequently spanning multiple \emph{jobs} carried out under the control of a \emph{workflow manager} process. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
+\item \declareterm{rank}\emph{rank} refers to the ordinal number (starting from zero) assigned to a process within a set of peer processes.  For example, the set of processes could be the members of a job, a compute node, or an application execution. \emph{Global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
```

Comment:
Process -> I am not sure this is a good idea to tie this processes, which I assume are OS processes. This is an implementation detail (sure, common at the moment, but with new architectures, accelerators, ... not a fixed thing). One way around that would be to define the term "PMIx processes" and then explicitly state that those are the managing granularity of PMIx, but not guaranteed to map 1:1 to OS processes (similar to the trick we had to do - unfortunately after the fact - in MPI)

Reply:
Per my earlier comment, I think this is a good idea, but will ikely be a much broader scope change than just Ch. 2.
I suggest we work on another ticket just for broadening the definition that will chase this PR. So the final result is that both get into v5, but by separating them we can more easily contain the discussion.

Reply:
That is probably the best way, but it should really be done at the same time as not to start making assumptions. We had this in MPI and it caused troubles. I was just pointing this out here to avoid the same issues, especially as this seems more severe here, as PMIx also targets (or should target) heterogenous systems.

Reply:
> I suggest we work on another ticket just for broadening the definition that will chase this PR.

Created an issue to track this: https://github.com/pmix/pmix-standard/issues/262